Duplicati/Library/Common/IO/SystemIOWindows.cs

```diff
@@ -288,20 +288,47 @@ public string GetSymlinkTarget(string file)
             return null;
         }
 
+        private static T TryNativeFirst<T>(Func<string, T> nativeIOFunc, Func<string, T> alternativeIOFunc, string path)
+        {
+            if (!IsPathTooLong(path))
+                try { return nativeIOFunc(path); }
+                catch (System.IO.PathTooLongException) { }
+                catch (System.ArgumentException) { }

```

Comment:
Can I ask why we're ignoring the `ArgumentException` here?  Are there cases where `Directory.EnumerateFiles` will throw this, but the AlphaFS version is able to handle the provided path?

Reply:
Thanks for the explanation.  That makes sense.  If we do decide to drop the extra `catch`, we can do so in a separate pull request.  This way, we won't pollute this one with too many unrelated changes.

Reply:
Actually, it looks like there was a case where the .NET library threw an `ArgumentException` when it should have thrown a `PathTooLongException`.  Not sure if this is still the case, but this might be worth documenting in the code to clarify the extra `catch`.

See https://github.com/duplicati/duplicati/issues/1451#issuecomment-137708490